### ISSUE TITLE
CRITICAL LibUV Bug Kills whole server before route execution

### ISSUE THREAD

**Issue by AjaniBilby** [remix-run/remix#10241]
> Edited to summarise so far

There are potentially two issues, one which can only take down a development server, and once which can take down a production remix.js * express.js server (haven't tested on other adapters).

Both issues trigger the same libuv assertion which is not handled anywhere causing the entire node instance to exit.
The lowest JS function which ends up calling the libuv function has not yet be found and may not be part of remix.js, but it does affect remix.js servers so I'll keep this issue open until the true fault is found.

Note, this crash is causing the server to close so fast that the console.log buffers are not flushed which makes finding the culpret function quite annoying.

### Reproduction

1. Clone [AjaniBilby/remix-lib-uv-assert](https://github.com/AjaniBilby/remix-lib-uv-assert)
2. Install deps `npm i`

#### Malformed URL
1. Start the development server `npm run dev`
2. Make the following curl request

```bash
curl -i -X POST \
   -d \
'{}' \
'http://localhost:3000/api/address/validate/%E1%9C%84%C8%BAy%F0%90%9E%B2:%F0%9E%A2%A2%F0%98%B4%87%F0%90%80%80'%C2%A53%CC%9E[%3Ci$?%F0%9C%BC%86%DF%8E%EF%BF%BDC%EF%BF%BD%E2%80%97%F0%91%84%BC%E3%81%B6%27'
```

Fails an assertion in what I assume is libuv, however it appears to fail so rapidly that morgan doesn't get to log the request.
```
App listening on http://localhost:3000
Assertion failed: w_target_len == 0, file c:\ws\deps\uv\src\idna.c, line 408
``` 

#### Malformed Packet
1. Start the production server `npm run start` or dev `npm run dev
2. Send the corrupted packet `node kill.js`

```bash
> cross-env NODE_ENV=production node ./server.js

App listening on http://localhost:3000
Assertion failed: w_target_len == 0, file c:\ws\deps\uv\src\idna.c, line 408
```

### System Info

```shell
System:
    OS: Windows 11 10.0.22631
    CPU: (22) x64 Intel(R) Core(TM) Ultra 7 155H
    Memory: 4.30 GB / 15.74 GB
  Binaries:
    Node: 22.2.0 - C:\Program Files\nodejs\node.EXE
    npm: 10.9.0 - C:\Program Files\nodejs\npm.CMD
  Browsers:
    Edge: Chromium (128.0.2739.42)
    Internet Explorer: 11.0.22621.3527
```

**Comment by AjaniBilby** [comment]
I was having the issue occur before on production and dev, but the repo I linked only has it occuring on dev, which of course would be much less of a worry, and is possibly a vite problem if it's dev only.

**Comment by AjaniBilby** [comment]
[Affected LibUV Source](https://github.com/libuv/libuv/blob/v1.x/src/idna.c#L387-L409) seems to be a UTF-8 to UTF-16 conversion issue

**Comment by AjaniBilby** [comment]
Found an exact packet that will consistently kill a production remix.js express.js server
[source](https://github.com/AjaniBilby/remix-lib-uv-assert/blob/dcf6f0e9f0b870ab34cba7db1d62fae35a9b75a6/kill.js#L3-L21)

**Comment by AjaniBilby** [comment]
NVM, actually the `API-KEY` header might have been a red-herring.
It's up to the maintainers whether or not this issue should remain open as it's not an issue directly with remix.js, but does affect it quite spectacularly until patched.

**Comment by AjaniBilby** [comment]
This bug was patched in [libuv#4609](https://github.com/libuv/libuv/pull/4609) now it just needs to travel downstream to Node.
